pkg/cache/application.go

```diff
@@ -693,7 +676,25 @@ func (app *Application) handleReleaseAppAllocationAskEvent(event *fsm.Event) {
 			zap.String("appID", app.applicationID),
 			zap.String("taskID", taskID))
 	}
-	app.postAppReleased()
+}
+
+func (app *Application) handleAppTaskCompletedEvent(event *fsm.Event) {
+	eventArgs := make([]string, 2)
+	if err := events.GetEventArgsAsStrings(eventArgs, event.Args); err != nil {
+		log.Logger().Error("fail to parse event arg", zap.Error(err))
+		return
+	}
+	if app.GetApplicationState() != events.States().Application.Resuming {
+		return
+	}
```

Comment:
The state machine should have ensured this is only handled under `Resuming` state:

```
{
  Name: string(events.AppTaskCompleted),
  Src: []string{states.Resuming,
  Dst: states.Resuming
},
```
I think we can safely remove this.